lib/datadog/lambda/trace/ddtrace.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+# ddtrace is an optional dependency for the ruby package
+begin
+  require 'ddtrace'
+rescue LoadError
+  Datadog::Utils.logger.debug 'dd-trace unavailable'
+end
+
+module Datadog
+  # TraceListener tracks tracing context information
+  module Trace
```

Comment:
I would be careful with this kind of namespacing: although I don't think it causes a conflict today, there's a risk that `Datadog::Trace` is something that could have a conflict with `ddtrace`. As this looks like a kind of adapter, I would recommend namespacing this under Lambda e.g. something like `Datadog::Lambda::Trace`.

Reply:
That's a good point. I don't think we have any publically documented functions in these namespaces, so it should be an easy update.

Reply:
I think I'll update this in a follow up PR, just to keep the diff on this file clean.